addons/binding/org.openhab.binding.chamberlainmyq/src/main/java/org/openhab/binding/chamberlainmyq/internal/HttpUtil.java

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.chamberlainmyq.internal;
+
+import static org.openhab.binding.chamberlainmyq.ChamberlainMyQBindingConstants.USERAGENT;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.InputStreamContentProvider;
+import org.eclipse.jetty.http.HttpField;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.util.B64Code;
+import org.eclipse.jetty.util.StringUtil;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Need to Subclass to override the USERAGENT ID to spoof the MyQ App
+ * Some common methods to be used in both HTTP-In-Binding and HTTP-Out-Binding
+ *
+ * @author Scott Hanson - Initial contribution
+ * @author Thomas Eichstaedt-Engelen
+ * @author Kai Kreuzer - Initial contribution and API
+ * @author Svilen Valkanov - replaced Apache HttpClient with Jetty
+ */
+public class HttpUtil {
+
+    private static Logger logger = LoggerFactory.getLogger(HttpUtil.class);
+
+    private static HttpClient client = new HttpClient(new SslContextFactory());
+
+    /**
+     * Executes the given <code>url</code> with the given <code>httpMethod</code>.
+     * Furthermore the <code>http.proxyXXX</code> System variables are read and
+     * set into the {@link HttpClient}.
+     *
+     * @param httpMethod the HTTP method to use
+     * @param url        the url to execute (in milliseconds)
+     * @param timeout    the socket timeout to wait for data
+     *
+     * @return the response body or <code>NULL</code> when the request went wrong
+     * @throws IOException when the request execution failed, timed out or it was interrupted
+     */
+    public static String executeUrl(String httpMethod, String url, int timeout) throws IOException {
```

Comment:
There are framework methods that do exactly this in core.net.

Reply:
I need to override the user-agent in the HTTP header. The default implementation does not support this.

Reply:
What do we need in core to support this?

Reply:
I had to add `client.setUserAgentField(new HttpField(HttpHeader.USER_AGENT, USERAGENT));` and I set the USERAGENT = "Chamberlain/3.73"; I thought maybe you could add a new variable to the class constructor or the method call to do this. I assume the variable would have to be defaulted to null or the old value to not break other stuff.

Reply:
But in the end the user-agent-field is an header, why can't you just set the header?